scala-thoughts.md

```diff
@@ -0,0 +1,39 @@
+The first thing I would say about Scala is that there is too much syntax. It
+seems that there are several different ways of doing every simple thing, and
+which syntax you should use in each place is very unclear. For example, I'm
+still unclear what the difference between the following statements are:
+
+    f g x
+    f.g(x)
+    f(g(x))
+
+Since Scala likes implementing DSLs so much, it appears that lots of packages
+are more DSL than API, which usually makes the API easier to read, but harder
+to learn and write since you have to learn a largely new syntax over again.
```

Comment:
That's a good point - the flexibility in writing the code can make it harder for users.  I suspect a good style guide could solve this issue.